ethportal-peertest/src/jsonrpc.rs

```diff
@@ -56,49 +36,113 @@ fn validate_endpoint_response(method: &str, result: &Value) {
             assert!(local_key.as_str().unwrap().contains("0x"));
             assert!(result.get("buckets").unwrap().is_array());
         }
-        "eth_blockNumber" => {
-            assert!(result.is_string());
-            assert!(result.as_str().unwrap().contains("0x"));
-        }
-        "portalHistory_dataRadius" => {
+        "portal_historyRadius" => {
             assert_eq!(result.as_str().unwrap(), U256::from(u64::MAX).to_string());
         }
-        "portalState_dataRadius" => {
+        "portal_stateRadius" => {
             assert_eq!(result.as_str().unwrap(), U256::from(u64::MAX).to_string());
         }
+        "portal_historyPing" => {
+            assert!(result.is_string());
+            assert!(result.as_str().unwrap().contains("data_radius"));
+            assert!(result.as_str().unwrap().contains("payload"));
+        }
+        "portal_statePing" => {
+            assert!(result.is_string());
+            assert!(result.as_str().unwrap().contains("data_radius"));
+            assert!(result.as_str().unwrap().contains("payload"));
+        }
         _ => panic!("Unsupported endpoint"),
     };
     info!("{:?} returned a valid response.", method);
 }
 
 impl JsonRpcEndpoint {
-    pub fn all_endpoints() -> Iter<'static, Self> {
-        ALL_ENDPOINTS.iter()
+    pub fn all_endpoints(target_node: String) -> Vec<JsonRpcEndpoint> {
```

Comment:
Don't think we need to address #151 here, but it would be nice to enforce some exhaustive list, perhaps with some clever use of a `match` statement on `TrinEndpoint`. A nice/simple way to do that is not clear to me, so I haven't suggested it here, but it's probably something we should address before we add too many more endpoints to the test harness.

Reply:
Yup, definitely agree. Though, it needs to be done in a way that we can also exclude certain tests (eg. jsonrpc endpoints that call to infura which we don't always want to run). Tracking this in #185